proposals/0000-specify-dump-json.md

```diff
@@ -0,0 +1,155 @@
+
+## Background
+
+In the prior implementation of GHC, errors were passed around as `SDoc`s (for Structured Documents).
+These `SDoc`s are simply a text encoding of the errors with some formatting features, such as nice alignment.
+By transforming the errors into `SDoc`s at the moment of creation, a lot of type-level information is lost,
+causing a headache for GHC developers and GHC API users.
+To improve on this problem, the [journey to represent errors as structured values has begun](https://gitlab.haskell.org/ghc/ghc/-/wikis/Errors-as-(structured)-values ). 
+In summary, this means passing errors around as a specific type which contains the relevant information separated into different fields,
+rather than combining it all into one single `SDoc`.
+
+Another change that has been made is the introduction of [error codes](https://gitlab.haskell.org/ghc/ghc/-/issues/21684) to uniquely identify each possible error output from GHC. 
+
+These features together serve to make error consumption a breeze for GHC API users, but what about those that would like to consume errors without utilizing GHC's API?
+
+## Problem Statement
+
+The fundamental issue is that the structured errors are only presented to users who leverage GHC as a library in their application. If instead, you desire to consume error messages without tapping into the internals of GHC, your best bet is to parse the plain text errors printed to stdout by GHC. This poses two important problems. 
+
+The first is that this is an _unnecessary_ pain. The errors are now passed as structured values internally, but the users are given a fairly unstructured value to parse themselves, leading to the requirement of always having to write a parser whenever consumption is desired.
+
+The second problem is that this plain text output is subject to change. Messages are often changed to make the content clearer, but each change bubbles into a new requirement for the hand-written plain-text parser of error messages. This problem is mitigated a bit by the above-mentioned new inclusion of error codes, but even these error codes must be parsed from a non-standard structured output. 
+
+The requirements to solve this problem are to enable the consumption of GHC's diagnostic message in a structured representation that closely matches the internal representation without requiring users to utilize the internals of GHC. The most reasonable solution is to use a compiler flag to dump the structured messages into a versioned JSON output, which is described by a JSON Schema, to standardize the expected output. The hope is that the community can reach a general consensus on the requirements imposed by such a JSON schema. While it will not be too costly to change the JSON schema as new requirements come up (simply increment the schema version), the ideal situation is to present here an opportunity to provide input on the current schema and receive constructive criticism.
+
+## Prior Art and Related Efforts
```

Comment:
I think a useful point of comparison is the LSP spec, which defines a type for `Diagnostic`s. While there's certainly no need to mimic what LSP (it has plenty of problems), it might be useful for inspiration.

Reply:
I actually liked the way LSP presented spans better, and altered the schema to reflect that. Thanks for sharing!